docs/arch/scheduler.md

```diff
@@ -0,0 +1,74 @@
+# Scheduler
+
+## Contributors
+
+- DE MARTINO Giada Flora
+- DURAT Mathias
+- PLANCHE Benoit
+- TCHILINGUIRIAN Théo
+
+
+## Lexicon
+
+- Step: a step is a CI job unit.
+- Agent: an agent is a computing node registered with the scheduler. They constitute a pool of available computing resources.
+- Scheduling: selection of an agent on which to execute a received step.
+- Step execution stage: the state of the execution of a step (running, sucessful, failed).
+
+
+## The "What", "Why" and "How" of the scheduler.
+
+### What?
+
+The scheduler receives a stream of CI steps (CI job units) as its main input. It also tracks a set of CI agents (registered computing nodes) that provide a dynamic resource pool, where the CI steps will be executed.
```

Comment:
I think you will not receive steps, you will only receive jobs that contain steps.

Reply:
Weird for me. So you will get a stream of random steps and you'll have to know which one go with which one ? For me you'll definitly will receive an action/job that contain:
- configuration part
- execution part 
And that is the smallest object you'll deal with

Reply:
yes it was just a misunderstanding from the start